### PR TITLE
return early if CNI state exist and add some style fixes

### PR DESCRIPTION
**Pull request by ecigar13**
**Reason for Change**:

- Fixed some items in this PR: https://github.com/Azure/azure-container-networking/pull/1392
- Fixed some linting issues
- Remove unused error messages, renamed

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests

**Notes**:

### CODE REVIEW

File: ipam/api.go [hunk 1]
```diff
@@ -9,21 +9,20 @@ import (
...
 	// Error responses returned by AddressManager.
-	errInvalidAddressSpace     = fmt.Errorf("Invalid address space")
-	errInvalidPoolId           = fmt.Errorf("Invalid address pool")
-	errInvalidAddress          = fmt.Errorf("Invalid address")
-	errInvalidScope            = fmt.Errorf("Invalid scope")
-	errInvalidConfiguration    = fmt.Errorf("Invalid configuration")
-	errAddressPoolExists       = fmt.Errorf("Address pool already exists")
-	errAddressPoolNotFound     = fmt.Errorf("Address pool not found")
-	errAddressPoolInUse        = fmt.Errorf("Address pool already in use")
-	errAddressPoolNotInUse     = fmt.Errorf("Address pool not in use")
```
Comment (tamilmani1989) [inline ipam/api.go:20]: is these errors not used ?

File: ipam/api.go [hunk 2]
```diff
@@ -9,21 +9,20 @@ import (
...
+	errInvalidConfiguration = fmt.Errorf("Invalid configuration")
+	errAddressPoolExists    = fmt.Errorf("Address pool already exists")
+	errAddressPoolNotFound  = fmt.Errorf("Address pool not found")
+	errAddressExists        = fmt.Errorf("Address already exists")
+	errAddressNotFound      = fmt.Errorf("Address not found")
+	errAddressInUse         = fmt.Errorf("Address already in use")
+	errNoAvailableAddresses = fmt.Errorf("No available addresses")
+
+	// Exportable errors returned by AddressManager
+	ErrNoAvailableAddressPools = fmt.Errorf("no available address pools")
```
Comment (tamilmani1989) [inline ipam/api.go:25]: move to separate var block

File: ipam/api.go (hunk 1 above)
Comment (ecigar13) [reply ipam/api.go:20]: They are not. I don't know how they passed the linting to get to master branch.

Comment (tamilmani1989) [reply ipam/api.go:20]: linter was setup later.

File: ipam/api.go (hunk 2 above)
Comment (ecigar13) [reply ipam/api.go]: Done

**Comment by rbtr** [comment]
can you update the title to be descriptive of the change? [it becomes the commit message on merge](https://github.com/Azure/azure-container-networking/commits/master) so it is important that it provides immediately useful information.

**Review by rbtr — APPROVED** [review]
lgtm

File: cni/network/invoker_azure.go [hunk 3]
```diff
@@ -102,6 +102,10 @@ func (invoker *AzureIPAMInvoker) deleteIpamState() {
 		return
 	}
 
+	if cniStateExists {
```
Comment (aegal) [inline cni/network/invoker_azure.go:105]: Nit: with this we can remove the below !cniStateExists on line 115

File: cni/network/invoker_azure.go [hunk 4]
```diff
@@ -102,6 +102,10 @@ func (invoker *AzureIPAMInvoker) deleteIpamState() {
...
+		return
```
Comment (aegal) [inline cni/network/invoker_azure.go:106]: I think we should put a comment here explaining why we return if cniStateExists. Even for folks that have seen this code path before, it's not immediately obvious.

**Pull request merged**